packages/cli/src/commands/deploy/spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Question, Arg, Option, ArgsAndOpts } from '../../register-command';
+import { MethodArg } from '../../prompts/prompt';
+
+export const name = 'deploy';
+export const description = 'deploy a contract instance';
+
+export const args: Arg[] = [
+  {
+    name: 'contract',
+    async prompt(): Promise<Question> {
+      const choices = await import('../../prompts/choices');
+
+      // TODO: Make this include contracts from _unlinked_ dependencies.
+      const contracts = choices.contracts('all');
+
+      return {
+        message: 'Pick a contract to deploy',
+        choices: contracts,
+        validate: value => contracts.includes(value),
+      };
+    },
+  },
+  {
+    name: 'arguments',
+    variadic: true,
+    async prompt(argsAndOpts: ArgsAndOpts): Promise<Question[]> {
+      const { fromContractFullName } = await import('../../utils/naming');
+      const { default: ContractManager } = await import('../../models/local/ContractManager');
+      const { argLabelWithIndex } = await import('../../prompts/prompt');
+      const { parseArg, getSampleInput } = await import('../../utils/input');
+
+      const contractName = argsAndOpts.contract as string;
+
+      const { package: packageName, contract: contractAlias } = fromContractFullName(contractName);
+      const contract = new ContractManager().getContractClass(packageName, contractAlias);
+      const constructorInputs: MethodArg[] = contract.schema.abi.find(f => f.type === 'constructor')?.inputs ?? [];
+
+      return constructorInputs.map((arg, index) => {
+        const placeholder = getSampleInput(arg);
+        const validationError = placeholder
+          ? `Enter a valid ${arg.type} such as: ${placeholder}`
+          : `Enter a valid ${arg.type}`;
+
+        return {
+          message: `${argLabelWithIndex(arg, index)}:`,
+          validate: (value: string) => {
+            try {
+              parseArg(value, arg);
+              return true;
+            } catch (err) {
+              return false;
+            }
+          },
+          validationError,
```

Comment:
I'm curious: why a static property instead of a message generated within the `validate` function, so it can reuse `value` and any calculations done by the validation?

Reply:
I didn't want to build a more general solution without seeing the need for it, and this separation felt more intuitive to me initially. Let's wait and see if this is enough?

Reply:
I've changed my mind on this.